### PR TITLE
Fix 1-pixel flickering between tiles on dark styles

### PR DESCRIPTION
**Pull request by mpulkki-mapbox**
This pull request fixes the flickering issue between tiles (https://github.com/mapbox/mapbox-gl-js/issues/11858) by extruding vertices of globe tiles slightly towards the neighbouring tiles. Based on empirical testing I found `0.5` pixels to be enough to stitch seams on all of my test devices using different pixel ratios. It should be noted that the texture sampling is unaffected even though tiles are scaled up by a tiny amount. Any possible extra fragments will have their uvs clamped to range `[0, 1]`.

Fixes https://github.com/mapbox/mapbox-gl-js/issues/11858

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [ ] include before/after visuals or gifs if this PR includes visual changes
 - [x] write tests for all new functionality
 - [ ] document any changes to public APIs
 - [ ] post benchmark scores
 - [x] manually test the debug page
 - [ ] tagged `@mapbox/map-design-team` `@mapbox/static-apis` if this PR includes style spec API or visual changes
 - [ ] tagged `@mapbox/gl-native` if this PR includes shader changes or needs a native port
 - [x] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'
 - [x] add an entry inside this element for inclusion in the `mapbox-gl-js` changelog: `<changelog>Fix pixel flickering between tiles on darker styles in globe view.</changelog>`

### CODE REVIEW

File: src/geo/projection/globe_util.js [hunk 1]
```diff
@@ -499,17 +499,40 @@ export function globeUseCustomAntiAliasing(painter: Painter, context: Context, t
...
-    const tileZoom = 1 << id.z;
-    return [0, x, tileZoom, y, 0, id.y, n, w, S];
+
+    // Construct transformation matrix for converting tile coordinates into LatLngs
+    const tiles = 1 << id.z;
+    const tileWidth = e - w;
+    const tileHeight = n - s;
+    const toUv = 1.0 / GLOBE_VERTEX_GRID_SIZE;
+    const x = tileWidth * toUv;
+    const y = -tileHeight / GLOBE_LATITUDINAL_GRID_LOD_TABLE[latitudinalLod];
```
Comment (akoylasar) [inline src/geo/projection/globe_util.js:514]: how about `toUvX` and  we could also refactor `1 / GLOBE_LATITUDINAL_GRID_LOD_TABLE(..)` to `toUvY`

File: src/geo/projection/globe_util.js [hunk 2]
```diff
@@ -499,17 +499,40 @@ export function globeUseCustomAntiAliasing(painter: Painter, context: Context, t
...
+
+    if (id.z > 0) {
+        // Add slight padding to patch seams between tiles.
+        const pixelPadding = 0.5;
+        const padding = pixelPadding * 360.0 / worldSize;
+
+        const xScale = padding / tileWidth + 1;
+        const yScale = padding / tileHeight + 1;
+        const padMatrix = [xScale, 0, 0, 0, yScale, 0, -0.5 * padding / x, 0.5 * padding / y, 1];
+
```
Comment (akoylasar) [inline src/geo/projection/globe_util.js:526]: The code here has now become more complicated, perhaps we could add a comment along the lines of:
```
// latLon = gridMatrix * (p_grid + scale * (p_grid - grid_center)) which implies
// latLon = grdiMatrix * (p_grid + scale * p_grid - scale * grid_center)
// and the part in parentheses is another transformation which can be represented by another matrix as follows:
// latLon = gridMatrix * padMatrix * p_grid
```

**Review by akoylasar — APPROVED** [review]
LGTM + some minor comments

File: src/geo/projection/globe_util.js (hunk 1 above)
Comment (mpulkki-mapbox) [reply src/geo/projection/globe_util.js:514]: I think it's unnecessary to have a separate variable `toUv` anymore

File: src/geo/projection/globe_util.js (hunk 2 above)
Comment (mpulkki-mapbox) [reply src/geo/projection/globe_util.js:526]: This is fairly trivial transformation matrix composition. Perhaps I could add a comment or two about coordinate transformations

Comment (akoylasar) [reply src/geo/projection/globe_util.js:523]: indeed I was trying to capture the transformation from spaces as I think that might not be trivial at all for a person encountering it for the first time.

**Review by mourner — APPROVED** [review]
Looks good to me! I wonder if we should add a more specific render test that has white gaps in pre-PR rendering?

**Comment by mpulkki-mapbox** [comment]
> Looks good to me! I wonder if we should add a more specific render test that has white gaps in pre-PR rendering?

The greenish looking render test `globe-tile-padding` is supposed to catch any gaps between tiles. It does not pass (at least locally) without this change.

**Pull request merged**